sdk/clientcore/core/src/main/java/io/clientcore/core/util/serializer/XmlSerializer.java

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.serializer;
+
+import io.clientcore.core.implementation.TypeUtil;
+import io.clientcore.core.serialization.xml.XmlReader;
+import io.clientcore.core.serialization.xml.XmlSerializable;
+import io.clientcore.core.serialization.xml.XmlWriter;
+import io.clientcore.core.util.ClientLogger;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+/**
+ * Class providing basic XML serialization and deserialization methods.
+ * <p>
+ * The implementation of this class is based on the usage of {@link XmlReader} and {@link XmlWriter}.
+ * <p>
+ * The deserialization methods only work with models implementing {@link XmlSerializable}. Or, in code terms, types that
+ * provide a static factory method {@code fromXml(XmlReader)}.
+ * <p>
+ * The serialization methods only work with complex types that implement {@link XmlSerializable}.
+ */
+public class XmlSerializer implements ObjectSerializer {
```

Comment:
Does this function as default implementation for XML Serialization. How different is this from SimpleXMLSerializable, can the both be merged to be DefaulXMLSerializable?

Reply:
`SimpleXmlSerializable` is an implementation of `XmlSerializable`, so it's a model type. This is the XML serializer implementation of `ObjectSerializer`.